internal/metadata/ssh_key_parser.go

```diff
@@ -0,0 +1,58 @@
+package metadata
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/digitalocean/droplet-agent/internal/sysaccess"
+	"regexp"
+	"strings"
+)
+
+const (
+	osUserRegex = `-os_user=(\S+)`
+)
+
+// SSHKeyParser is used for parsing different type of SSH Keys from metadata
+type SSHKeyParser struct {
+	pubKeyOSUserRegex *regexp.Regexp
+}
+
+// NewSSHKeyParser returns a new SSH Key parser
+func NewSSHKeyParser() *SSHKeyParser {
+	return &SSHKeyParser{
+		pubKeyOSUserRegex: regexp.MustCompile(osUserRegex),
+	}
+}
+
+// FromPublicKey parses a string public key and attempts to convert it to a SSHKey object
+func (p *SSHKeyParser) FromPublicKey(key string) (*sysaccess.SSHKey, error) {
+	ret := &sysaccess.SSHKey{
+		PublicKey: strings.Trim(key, " \t\r\n"),
+		Type:      sysaccess.SSHKeyTypeDroplet,
+	}
+	match := p.pubKeyOSUserRegex.FindAllStringSubmatch(key, -1)
+	if len(match) != 0 {
+		// trying to find "-os_user" flag from the key
+		// if not presented, default user will be used
+		// if multiple os_user flags presented, the last one will be used
```

Comment:
I think there should be some documentation or link to rationale about why the last one is used (say, vs. the first one).

Reply:
That's a good point, I'll add it to the comment. It's because the backend service will always append it to the end of the key.